### ISSUE TITLE
Lỗi cài requirements torch khi sử dụng pypy

### ISSUE THREAD

**Issue by thangnd85** [undertheseanlp/underthesea#387]
Lỗi cài requirements torch khi sử dụng pypy
Tương tự lỗi với debian buster mới.

**Comment by rain1024** [comment]
@thangnd85 Bạn nêu rõ môi trường của bạn (hệ điều hành, phiên bản python,..)

Câu lệnh bạn sử dụng và chi tiết lỗi khi cài đặt nhé

**Comment by thangnd85** [comment]
Mình cài được với bản cũ hơn rồi. Để mình cài lại với bản 1.3.1 rồi báo lại team nhé.
Thanks :)

**Comment by rain1024** [comment]
@thangnd85 tuyệt vời !!!

**Comment by thangnd85** [comment]
Log đây team nhé
` ERROR: Command errored out with exit status 1:
   command: /home/pi/vinahome/venv/bin/pypy3-c /home/pi/vinahome/venv/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-h30kad06/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'Cython>=0.28.5' 'numpy==1.13.3; python_version=='"'"'3.6'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system!='"'"'AIX'"'"' and platform_python_implementation=='"'"'CPython'"'"'' 'numpy==1.14.0; python_version=='"'"'3.6'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system!='"'"'AIX'"'"' and platform_python_implementation!='"'"'CPython'"'"'' 'numpy==1.16.0; python_version=='"'"'3.6'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.7'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.19.0; python_version=='"'"'3.6'"'"' and platform_machine=='"'"'aarch64'"'"'' 'numpy==1.19.0; python_version=='"'"'3.7'"'"' and platform_machine=='"'"'aarch64'"'"'' 'numpy==1.19.0; python_version=='"'"'3.8'"'"' and platform_machine=='"'"'aarch64'"'"'' 'numpy==1.14.5; python_version=='"'"'3.7'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.17.3; python_version=='"'"'3.8'"'"' and platform_machine!='"'"'aarch64'"'"'' 'numpy==1.19.3; python_version=='"'"'3.9'"'"'' 'scipy>=0.19.1'
       cwd: None
  Complete output (21 lines):
  Ignoring numpy: markers 'python_version == "3.6" and platform_machine != "aarch64" and platform_system != "AIX" and platform_python_implementation == "CPython"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.6" and platform_machine != "aarch64" and platform_system == "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.7" and platform_machine != "aarch64" and platform_system == "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.6" and platform_machine == "aarch64"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.7" and platform_machine == "aarch64"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.8" and platform_machine == "aarch64"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.7" and platform_machine != "aarch64" and platform_system != "AIX"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.8" and platform_machine != "aarch64"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.9"' don't match your environment
  Collecting setuptools
    Using cached setuptools-51.1.2-py3-none-any.whl (784 kB)
  Collecting wheel
    Using cached wheel-0.36.2-py2.py3-none-any.whl (35 kB)
  Collecting Cython>=0.28.5
    Using cached Cython-0.29.21-py2.py3-none-any.whl (974 kB)
  Collecting numpy==1.14.0
    Downloading numpy-1.14.0.zip (4.9 MB)
  Collecting scipy>=0.19.1
    Downloading scipy-1.5.4.tar.gz (25.2 MB)
  ERROR: Could not install packages due to an EnvironmentError: [Errno 28] Không có đủ chỗ trống trên thiết bị
  
  ----------------------------------------
ERROR: Command errored out with exit status 1: /home/pi/vinahome/venv/bin/pypy3-c /home/pi/vinahome/venv/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-h30kad06/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'Cython>=0.28.5' 'numpy==1.13.3; python_version=='"'"'3.6'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system!='"'"'AIX'"'"' and platform_python_implementation=='"'"'CPython'"'"'' 'numpy==1.14.0; python_version=='"'"'3.6'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system!='"'"'AIX'"'"' and platform_python_implementation!='"'"'CPython'"'"'' 'numpy==1.16.0; python_version=='"'"'3.6'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.16.0; python_version=='"'"'3.7'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system=='"'"'AIX'"'"'' 'numpy==1.19.0; python_version=='"'"'3.6'"'"' and platform_machine=='"'"'aarch64'"'"'' 'numpy==1.19.0; python_version=='"'"'3.7'"'"' and platform_machine=='"'"'aarch64'"'"'' 'numpy==1.19.0; python_version=='"'"'3.8'"'"' and platform_machine=='"'"'aarch64'"'"'' 'numpy==1.14.5; python_version=='"'"'3.7'"'"' and platform_machine!='"'"'aarch64'"'"' and platform_system!='"'"'AIX'"'"'' 'numpy==1.17.3; python_version=='"'"'3.8'"'"' and platform_machine!='"'"'aarch64'"'"'' 'numpy==1.19.3; python_version=='"'"'3.9'"'"'' 'scipy>=0.19.1' Check the logs for full command output.`

**Comment by thangnd85** [comment]
Hệ điều hành: armbian bulleyes (debian 11)
pypy3 (python3.6)
Không build được scikit learn.
Sau khi cài scikit-learn bằng python3-sklearn, cài underthesea thì lỗi như sau
`ERROR: Could not find a version that satisfies the requirement torch<=1.5.1,>=1.1.0 (from underthesea) (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2)
ERROR: No matching distribution found for torch<=1.5.1,>=1.1.0 (from underthesea)`

**Comment by rain1024** [comment]
**1. môi trường của bạn (hệ điều hành, phiên bản python,..)**

Bạn có dùng [anaconda](https://docs.anaconda.com/anaconda/install/) hay [venv](https://docs.python.org/3.6/tutorial/venv.html) không?

Hiện tại, mình đã cài đặt underthesea qua google colab vẫn cài đặt được bình thường. [[link google colab]](https://colab.research.google.com/drive/1SPMd7gu_bj9YkhRCAshU5jvAZx4Z01wU?usp=sharing)

**2. Chi tiết câu lệnh bạn sử dụng**

Bạn cài đặt underthesea bằng những câu lệnh nào? Có phải 2 câu lệnh này không?

```
pip install python3-sklearn 
pip install underthesea 
```

Bạn thử sử dụng [anaconda](https://docs.anaconda.com/anaconda/install/) để tạo môi trường độc lập, sau đó cài đặt underthesea thử xem 

```
conda create -n lab_underthesea python=3.6
conda activate lab_underthesea
pip install underthesea
```

**3.Sau khi cài scikit-learn bằng python3-sklearn, cài underthesea thì lỗi như sau**

`ERROR: Could not find a version that satisfies the requirement torch<=1.5.1,>=1.1.0 (from underthesea) (from versions: 0.1.2, 0.1.2.post1, 0.1.2.post2)
ERROR: No matching distribution found for torch<=1.5.1,>=1.1.0 (from underthesea)`

Mình nghĩ lỗi này là do phiên bản của python không tương tích với các phiên bản của torch `torch<=1.5.1,>=1.1.0 (from underthesea)`

Bạn thử sử dụng [anaconda](https://docs.anaconda.com/anaconda/install/) để tạo môi trường độc lập rồi cài đặt xem có lỗi này nữa không nhé.

**Comment by thangnd85** [comment]
Mình dùng venv, lỗi có thể do torch không hỗ trợ cho armhf.

**Comment by rain1024** [comment]
@thangnd85 

> Mình dùng venv, lỗi có thể do torch không hỗ trợ cho armhf.
cũng có thể do nguyên nhân này

Ngoài ra, mình thấy ở issue này (https://github.com/pytorch/pytorch/issues/10443), cũng có vài report về vấn đề cài đặt torch với pipenv, và sau đó cài đặt thành công với conda. Bạn thử sử dụng conda xem sao nhé

**Comment by thangnd85** [comment]
Conda thì cũng không hỗ trợ armhf nên phải quay về underthesea==1.2.3 thôi

**Comment by rain1024** [comment]
@thangnd85 

> Conda thì cũng không hỗ trợ armhf nên phải quay về underthesea==1.2.3 thôi

Vậy thì tiếc quá.

> Mình dùng venv, lỗi có thể do torch không hỗ trợ cho armhf.

Nếu thực sư torch không hỗ trợ cho armhf thì @thangnd85 có thể tạo issue ở dự án pytorch xem nhé, mình nghĩ team pytorch sẽ ghi nhận issue và tìm cách giải quyết trong tương lai

**Comment by krithika-arunachalam** [comment]
Facing similar issues, I have hugging face model downloaded using transformers=4.1.1. But while trying to use **underthesea** it says like this,

ERROR: underthesea 1.3.1 has requirement transformers<=3.5.1,>=3.5.0, but you'll have transformers 4.1.1 which is incompatible.

**Comment by rain1024** [comment]
@krithika-arunachalam I'm working in this issue. I will update transformers dependency. Please keep watching.

**Comment by rain1024** [comment]
Solved in underthesea v1.3.3
